gobblin-core/src/main/java/gobblin/source/extractor/hadoop/AvroFsHelper.java

```diff
@@ -139,14 +69,14 @@ public InputStream getFileStream(String path) throws FileBasedHelperException {
   public Schema getAvroSchema(String file) throws FileBasedHelperException {
     DataFileReader<GenericRecord> dfr = null;
     try {
-      if (state.getPropAsBoolean(ConfigurationKeys.SHOULD_FS_PROXY_AS_USER,
+      if (this.getState().getPropAsBoolean(ConfigurationKeys.SHOULD_FS_PROXY_AS_USER,
           ConfigurationKeys.DEFAULT_SHOULD_FS_PROXY_AS_USER)) {
         dfr =
-            new DataFileReader<GenericRecord>(new ProxyFsInput(new Path(file), this.fs),
```

Comment:
I'm going to pass on this as well.  This is legacy code and a refactoring that I'm not prepared to test right now.